modules/meteroid/crates/diesel-models/src/query/customers.rs

```diff
@@ -154,6 +155,53 @@ impl CustomerRow {
             .attach_printable("Error while inserting customer batch")
             .into_db_result()
     }
+
+    pub async fn select_for_update(conn: &mut PgConn, id: Uuid) -> DbResult<CustomerRow> {
+        use crate::schema::customer::dsl as c_dsl;
+        use diesel_async::RunQueryDsl;
+
+        let query = c_dsl::customer.for_no_key_update().filter(c_dsl::id.eq(id));
+
+        log::debug!("{}", debug_query::<diesel::pg::Pg, _>(&query).to_string());
+
+        query
+            .first(conn)
+            .await
+            .attach_printable("Error while selecting for update customer by id")
+            .into_db_result()
+    }
+
+    pub async fn update_balance(conn: &mut PgConn, id: Uuid, delta_cents: i32) -> DbResult<()> {
+        use crate::schema::customer::dsl as c_dsl;
+        use diesel_async::RunQueryDsl;
+
+        let query = diesel::update(c_dsl::customer)
+            .filter(c_dsl::id.eq(id))
+            .filter(c_dsl::balance_value_cents.add(delta_cents).le(0))
```

Comment:
I guess we want to filter only if >=0 so `.ge` no ?

Reply:
yep, you are right, fixed, thanks!